etc/generate-docs.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# usage: ./etc/generate-docs.sh [new version string]
+
+# exit if any command fails
+set -e
+
+if ! command -v jazzy > /dev/null; then
+  gem install jazzy || { echo "ERROR: Failed to locate or install jazzy; please install yourself with 'gem install jazzy' (you may need to use sudo)"; exit 1; }
+fi
+
+version=${1}
+
+jazzy_args=(--clean
+            --author "Neal Beeken, Nellie Spektor, Patrick Freed, and Kaitlin Mahar" 
+            --readme "etc/docs-main.md" 
+            --author_url https://github.com/mongodb/swift-bson 
+            --github_url https://github.com/mongodb/swift-bson 
+            --theme fullwidth 
+            --documentation "Guides/*.md" 
+            --github-file-prefix https://github.com/mongodb/swift-bson/tree/v${version} 
+            --module-version "${version}" 
+            --swift-build-tool spm)
+
+modules=( BSON )
+
+for module in "${modules[@]}"; do
+  args=("${jazzy_args[@]}"  --output "docs/${module}" --module "${module}" 
+        --root-url "https://mongodb.github.io/swift-bson/docs/${module}/")
```

Comment:
How do we actually publish the docs to this URL?

Reply:
the repository settings are configured such that GitHub Pages will automatically deploy the contents of the `/docs` folder on the `gh-pages` branch:
![image](https://user-images.githubusercontent.com/5976377/91334115-7e69d180-e79c-11ea-9eaa-da872610df4e.png)